OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/NfcOperationActivity.java

```diff
@@ -67,6 +68,14 @@ protected void onCreate(Bundle savedInstanceState) {
 
         getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
 
+        mInputParcel = getIntent().getParcelableExtra(EXTRA_CRYPTO_INPUT);
+
+        if (mInputParcel == null) {
+            // for compatibility when used from OpenPgpService
+            // (or any place other than CryptoOperationHelper)
+            mInputParcel = new CryptoInputParcel(mRequiredInput.mSignatureTime);
```

Comment:
The current implementation of `CryptoOperationHelper` always assigns a date to `mSignatureTime`. If it isn't a signing operation and `mRequiredInput.mSignatureTime` is null, the current date is used to instantiate. Should we change this behavior of `CryptoOperationHelper` and introduce a `null` value for `mSignatureTime` when it isn't a signing operation? As of now it's declared as `final`, we'd just have to make it add-able like the rest of the attributes.